documentation/rest-clients.md

```diff
@@ -0,0 +1,158 @@
+import ExampleClient from "@azure-rest/example-client";
+```
+
+Most clients require authenticating through their `credential` parameter. Depending on what authentication support your library is using, you can either authenticate with AAD or authenticate with an AzureKeyCredential.
+
+Additionally, most of our clients accept an `endpoint` parameter at initialization, usually a link to your own resource.
+
+### Authenticating with AAD
+Authenticating with AAD
+Depending on your library, our clients support authenticating with an Azure Active Directory (AAD) token credential. We always recommend using a credential type obtained from the `@azure/identity` library for AAD authentication. For this example, we use the most common DefaultAzureCredential.
+
+As an installation note, the `@azure/identity` library is not a dependency of this library. Please run `npm install @azure/identity` before using AAD authentication
+
+The following code snippet shows you how to authenticate with a DefaultAzureCredential.
+
+```typescript
+import ExampleClient from "@azure-rest/example-client";
+import { DefaultAzureCredential } from "@azure/identity";
+
+const client = ExampleClient("https://example.org/", new DefaultAzureCredential());
+```
+
+## 2. Send a request
+
+Once the client has been initialized we need to set a path to work with. For this, the REST client exposes 2 functions `path` and `pathUnchecked`
+
+### Path
+Path function takes a path as the first parameter and accepts any path defined in the Swagger. It also detects if the path needs a path parameter and requires them as positional parameters to `path`. Once the path is set users can access functions for all the supported verbs on that path
+
+```typescript
+import ExampleClient from "@azure-rest/example-client";
+import { DefaultAzureCredential } from "@azure/identity";
+
+const client = ExampleClient ("https://example.org/", new DefaultAzureCredential());
+const response = await client.path("/hello/{name}", "Brian").get();
+
+// response.body is strongly typed
+console.log(response.body);
+// {content: "Hello Brian"}
+```
+
+### PathUnchecked
+PathUnchecked function is similar to Path, it takes a path as the first parameter, this can be any arbitrary path. It also detects if the path needs a path parameter and requires them as positional parameters to `pathUnchecked`. Once the path is set users can access functions for any verb on that path.
+
+The main difference with `path` is that `pathUnchecked` doesn't have strongly typed payload, headers, or query parameters and has `any` as the response type.
+
+```typescript
+import ExampleClient from "@azure-rest/example-client";
+import { DefaultAzureCredential } from "@azure/identity";
+
+const client = ExampleClient ("https://example.org/", new DefaultAzureCredential());
+const response = await client.pathUnchecked("/hello/{name}", "Brian").get();
+
+// response.body is not strongly typed
+console.log(response.body);
+// {content: "Hello Brian"}
```

Comment:
The differences are a bit hard to notice in this, can we invent a case where it would be useful to use pathUnchecked? e.g., modifying the path slightly or using a http verb (e.g. HEAD) that isn't in the default list?